File: c/isogram/src/isogram.c
#include "isogram.h"
#include <ctype.h>

__attribute__((pure))
static inline int rank(const char c) {
  return !isalpha(c) ? -1 : tolower(c) - 'a';
}

__attribute__((pure))
bool is_isogram(const char word[]) {
  if (!word) { return false; }
  
  for (bool uniq[26] = { 0 }; *word;) {
    const int i = rank(*word++);
    if (i >= 0 && !(uniq[i] ^= 1)) {
      return false;
    }
  }
  return true;
}

File: csharp/isogram/Isogram.cs
using System.Linq;
using System.Text.RegularExpressions;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var parsed = Regex.Replace(word.ToLower(), @"[^a-z]", "");
        return parsed.Distinct().Count() == parsed.Length;
    }
}
